### PR TITLE
Add option to adjust job OOM score

### PR DESCRIPTION
**Pull request by koct9i**

### CODE REVIEW

File: yt/yt/server/node/exec_node/job.cpp [hunk 1]
```diff
@@ -987,6 +987,24 @@ bool TJob::ResourceUsageOverdrafted() const
...
+    if (reservedMemory > totalMemory) {
+	reservedMemory = totalMemory;
+    }
+
+    // OOM score is memory usage normalized to total memory size or cgroup memory limit.
+    // This adjustment allows to subtract score for reserved memory. The same formula is
+    // used in Kubernetes.
+    //
+    // TODO(khlebnikov): Use memory limit dedicated for user jobs cgroup.
+    return MaxOOMScore - (MaxOOMScore * reservedMemory + totalMemory - 1) / totalMemory;
```
Comment (gritukan) [inline yt/yt/server/node/exec_node/job.cpp:1005]: Can't this TODO be easily fixed by taking UserJobs limit from memory usage tracker?

Comment (k-pogorelov) [inline yt/yt/server/node/exec_node/job.cpp:1005]: I'm not sure it is a suitable formula.
Why does MaxOOMScore decrease from an increase in memory demand?

**Review by k-pogorelov — CHANGES_REQUESTED** [review]

File: yt/yt/server/node/exec_node/job.cpp [hunk 2]
```diff
@@ -987,6 +987,24 @@ bool TJob::ResourceUsageOverdrafted() const
     return TResourceHolder::GetResourceUsage().UserMemory > RequestedMemory_;
 }
 
+i64 TJob::GetOOMScoreAdjustment() const
+{
+    auto totalMemory = Bootstrap_->GetMemoryUsageTracker()->GetTotalLimit();
```
[the remark this answers is not in the document]
Comment (koct9i) [reply yt/yt/server/node/exec_node/job.cpp:992]: I don't think so. Thist must be common cgroup memory limit for all jobs.
Thiere is no such thing yet.

And more likely this should be part of NodeResouceManager.

[the remark this answers is not in the document]
Comment (k-pogorelov) [reply yt/yt/server/node/exec_node/job.cpp:992]: Yes, this must be common cgroup memory limit for all jobs. And this limit must be returned by JobResourceManager->GetResourceLimits().

**Comment by koct9i** [comment]
Later

**Pull request closed without merging**